src/NotepadNext/docks/FolderAsWorkspaceDock.cpp

```diff
@@ -18,20 +18,27 @@
 
 
 #include "FolderAsWorkspaceDock.h"
+
+#include <QFileSystemModel>
+#include <QMessageBox>
+
 #include "ApplicationSettings.h"
+#include "MainWindow.h"
 #include "ui_FolderAsWorkspaceDock.h"
 
-#include <QFileSystemModel>
+QString NEW_DIR_TEMPLATE("dir_%1");
 
 ApplicationSetting<QString> rootPathSetting{"FolderAsWorkspace/RootPath"};
 
-FolderAsWorkspaceDock::FolderAsWorkspaceDock(QWidget *parent) :
+FolderAsWorkspaceDock::FolderAsWorkspaceDock(MainWindow *parent) :
     QDockWidget(parent),
-    ui(new Ui::FolderAsWorkspaceDock),
-    model(new QFileSystemModel(this))
+    window(parent)
 {
+    ui = new Ui::FolderAsWorkspaceDock;
+    model = new QFileSystemModel(this);
```

Comment:
Any particular reason these couldn't be kept in the initializer list?

Reply:
There was warning about `-Wreorder` I don't get how to remove